linkerd/app/inbound/src/direct.rs

```diff
@@ -182,9 +191,7 @@ impl<N> Inbound<N> {
                                     // connection is a gateway connection. We check the _gateway
                                     // address's_ policy to determine whether the client is
```

Comment:
does this comment need to be updated to reflect that we don't actually check it here? i guess the wording is ambiguous enough that maybe this is fine, but it might be a little clearer.

maybe just changing it to "we *will* check..." is sufficient?

Reply:
there's another comment here that maybe should also be changed: https://github.com/linkerd/linkerd2-proxy/blob/debb9cb802e43d6ea946a97428f61cea6a73d505/linkerd/app/inbound/src/direct.rs#L158-L161
but i can't leave an inline comment on that one in the diff.